egs/wsj/s5/utils/data/perturb_speed_to_allowed_lengths.py

```diff
@@ -0,0 +1,298 @@
+#!/usr/bin/env python
+
+# Copyright     2017  Hossein Hadian
+# Apache 2.0
+
+
+""" This script perturbs speeds of utterances to force their lengths to some
+    allowed lengths spaced by a factor (like 10%)
+"""
+
+import argparse
+import os
+import sys
+import copy
+import math
+
+def get_args():
+    parser = argparse.ArgumentParser(description="""This script copies the 'srcdir'
+                                   data directory to output data directory 'dir'
+                                   while modifying the utterances so that there are
+                                   3 copies of each utterance: one with the same
+                                   speed, one with a higher speed (not more than
+                                   factor% faster) and one with a lower speed
+                                   (not more than factor% slower)""")
+    parser.add_argument('factor', type=float, default=12,
+                        help='Spacing (in percentage) between allowed lengths.')
+    parser.add_argument('srcdir', type=str,
+                        help='path to source data dir')
+    parser.add_argument('dir', type=str, help='output dir')
+    parser.add_argument('--coverage-factor', type=float, default=0.05,
+                        help="""Percentage of durations not covered from each
+                             side of duration histogram.""")
+    parser.add_argument('--frame-shift', type=int, default=10,
+                        help="""Frame shift in milliseconds.""")
+    parser.add_argument('--frame-length', type=int, default=25,
+                        help="""Frame length in milliseconds.""")
+    parser.add_argument('--frame-subsampling-factor', type=int, default=3,
+                        help="""Chain frame subsampling factor.
+                             See steps/nnet3/chain/train.py""")
+    parser.add_argument('--speed-perturb',  type=bool, default=True,
+                        help="""If false, no speed perturbation will occur, i.e.
```

Comment:
Type casting to bool will change "false" to True. You can instead type=str, choices=['true','false'] and then convert the string to bool yourself.